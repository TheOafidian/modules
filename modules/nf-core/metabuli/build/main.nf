
process METABULI_BUILD {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/metabuli:1.0.5--pl5321h6a68c12_1':
        'biocontainers/metabuli:1.0.5--pl5321h6a68c12_1' }"

    input:
    tuple val(meta), path(db)
    path accession2taxid, stageAs: 'taxonomy/acc2taxid'

    output:
    tuple val(meta), path("$prefix"), emit: db
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    acc2taxid = accession2taxid ? "${accession2taxid}" : "${db}/taxonomy/acc2taxid"
    """
    find ${db}/library -type f -name '*.fna' > library-files.txt

    metabuli \\
        build \\
        ${db} \\
        library-files.txt \\
        ${acc2taxid} \\
        --db-name ${prefix} \\
        --threads ${task.cpus} \\
        $args

    if [[ \$(basename ${db}) != "${prefix}" ]]; then
        mkdir -p ${prefix}
        mv ${db}/* ${prefix}
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metabuli: \$(metabuli version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir -p "$prefix"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metabuli: \$(metabuli version)
    END_VERSIONS
    """
}
