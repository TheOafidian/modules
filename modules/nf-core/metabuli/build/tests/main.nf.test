nextflow_process {

    name "Test Process METABULI_BUILD"
    script "../main.nf"
    process "METABULI_BUILD"
    config "./nextflow.config"
    tag "metabuli"
    tag "metabuli/build"
    tag "metabuli/add"
    tag "modules"
    tag "modules_nfcore"

    setup {
        run("METABULI_ADD") {
            script "modules/nf-core/metabuli/add/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test' ],
                        [
                        file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta', checkIfExists: true),
                        file(params.modules_testdata_base_path +'genomics/sarscov2/genome/proteome.fasta', checkIfExists: true),
                        ]
                    ]
                input[1] = file(params.modules_testdata_base_path +'genomics/sarscov2/metagenome/prot_names.dmp', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path +'genomics/sarscov2/metagenome/prot_nodes.dmp', checkIfExists: true)
                input[3] = [] // needs a merged.dmp
                input[4] = file(params.modules_testdata_base_path + 'delete_me/metabuli/acc2taxid', checkIfExists: true)
                """
            }
        }

    }

    test("sarscov2 protein_db") {

        when {
            process {
                """
                input[0] = METABULI_ADD.out.db
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot (process.out).match()},
            )
        }
    }

    test("sarscov2 - protein_db - acc2taxid") {

        when {
            process {
                """
                input[0] = METABULI_ADD.out.db
                input[1] = file(params.modules_testdata_base_path + 'delete_me/metabuli/acc2taxid', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot (process.out).match()},
            )
        }
    }

    test("sarscov2 protein_db stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = METABULI_ADD.out.db
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot (process.out).match()}
            )
        }
    }
}
