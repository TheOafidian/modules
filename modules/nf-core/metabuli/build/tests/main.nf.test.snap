{
    "sarscov2 protein_db": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "acc2taxid.map:md5,4eb8b3bde8a5cc1414afeeeab385e208",
                            "db.parameters:md5,481509eb7cddc0ec93fbcd61b7820616",
                            "diffIdx:md5,8ac88d03fabf6917698a02841e410b33",
                            "info:md5,ba18c68b0d90630fad4dfe02b5101b72",
                            [
                                "694009.fna:md5,e73599798195a519ba2565c3f0275b93"
                            ],
                            "split:md5,6306fc471e89174e65c71fd0cc69f780",
                            "taxID_list:md5,40064b0651b7d558820d6fc1014de0fd",
                            [
                                "acc2taxid:md5,4ae5ac3b865cb7bf301c3cc72153a7a8",
                                "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "names.dmp:md5,130f9132095562e09c732679c562f5e9",
                                "nodes.dmp:md5,c471c27a4ce85ae74d2c63633c9ce1e3"
                            ],
                            "taxonomyDB:md5,029a03295afff91c5d2a011c7e8b8db5",
                            "unmapped.txt:md5,b8c64bc37b7dc465fdaf61b3cd90c29a"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,acaa33b47a40006743150a1f9db99393"
                ],
                "db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "acc2taxid.map:md5,4eb8b3bde8a5cc1414afeeeab385e208",
                            "db.parameters:md5,481509eb7cddc0ec93fbcd61b7820616",
                            "diffIdx:md5,8ac88d03fabf6917698a02841e410b33",
                            "info:md5,ba18c68b0d90630fad4dfe02b5101b72",
                            [
                                "694009.fna:md5,e73599798195a519ba2565c3f0275b93"
                            ],
                            "split:md5,6306fc471e89174e65c71fd0cc69f780",
                            "taxID_list:md5,40064b0651b7d558820d6fc1014de0fd",
                            [
                                "acc2taxid:md5,4ae5ac3b865cb7bf301c3cc72153a7a8",
                                "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "names.dmp:md5,130f9132095562e09c732679c562f5e9",
                                "nodes.dmp:md5,c471c27a4ce85ae74d2c63633c9ce1e3"
                            ],
                            "taxonomyDB:md5,029a03295afff91c5d2a011c7e8b8db5",
                            "unmapped.txt:md5,b8c64bc37b7dc465fdaf61b3cd90c29a"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,acaa33b47a40006743150a1f9db99393"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T13:06:57.889836817"
    },
    "sarscov2 - protein_db - acc2taxid": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "acc2taxid.map:md5,4eb8b3bde8a5cc1414afeeeab385e208",
                            "db.parameters:md5,481509eb7cddc0ec93fbcd61b7820616",
                            "diffIdx:md5,8ac88d03fabf6917698a02841e410b33",
                            "info:md5,ba18c68b0d90630fad4dfe02b5101b72",
                            [
                                "694009.fna:md5,e73599798195a519ba2565c3f0275b93"
                            ],
                            "split:md5,6306fc471e89174e65c71fd0cc69f780",
                            "taxID_list:md5,40064b0651b7d558820d6fc1014de0fd",
                            [
                                "acc2taxid:md5,4ae5ac3b865cb7bf301c3cc72153a7a8",
                                "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "names.dmp:md5,130f9132095562e09c732679c562f5e9",
                                "nodes.dmp:md5,c471c27a4ce85ae74d2c63633c9ce1e3"
                            ],
                            "taxonomyDB:md5,8efd02b34af565a1a808e43bc239c75a",
                            "unmapped.txt:md5,b8c64bc37b7dc465fdaf61b3cd90c29a"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,acaa33b47a40006743150a1f9db99393"
                ],
                "db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "acc2taxid.map:md5,4eb8b3bde8a5cc1414afeeeab385e208",
                            "db.parameters:md5,481509eb7cddc0ec93fbcd61b7820616",
                            "diffIdx:md5,8ac88d03fabf6917698a02841e410b33",
                            "info:md5,ba18c68b0d90630fad4dfe02b5101b72",
                            [
                                "694009.fna:md5,e73599798195a519ba2565c3f0275b93"
                            ],
                            "split:md5,6306fc471e89174e65c71fd0cc69f780",
                            "taxID_list:md5,40064b0651b7d558820d6fc1014de0fd",
                            [
                                "acc2taxid:md5,4ae5ac3b865cb7bf301c3cc72153a7a8",
                                "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "names.dmp:md5,130f9132095562e09c732679c562f5e9",
                                "nodes.dmp:md5,c471c27a4ce85ae74d2c63633c9ce1e3"
                            ],
                            "taxonomyDB:md5,8efd02b34af565a1a808e43bc239c75a",
                            "unmapped.txt:md5,b8c64bc37b7dc465fdaf61b3cd90c29a"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,acaa33b47a40006743150a1f9db99393"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T13:08:00.235429024"
    },
    "sarscov2 protein_db stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,acaa33b47a40006743150a1f9db99393"
                ],
                "db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,acaa33b47a40006743150a1f9db99393"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-21T13:08:14.59465885"
    }
}